packages/flame/lib/src/effects2/rotate_effect.dart

```diff
@@ -0,0 +1,60 @@
+import 'effect_controller.dart';
+import 'transform2d_effect.dart';
+
+/// Rotate a component around its anchor.
+///
+/// Two constructors are provided:
+///   - [RotateEffect.by] will rotate the target by the specified `angle`
+///     relative to its orientation at the onset of the effect. For example,
+///     rotating by `angle = tau/4` will turn the component 90° clockwise
+///     relative to its initial direction;
+///   - [RotateEffect.to] will rotate the target to the fixed orientation
+///     specified by the `angle`. For example, rotating to `angle = tau/4` will
+///     turn the component to look East regardless of its initial bearing.
+///
+/// This effect applies incremental changes to the component's angle, and
+/// requires that any other effect or update logic applied to the same component
+/// also used incremental updates.
+class RotateEffect extends Transform2DEffect {
+  RotateEffect.by(double angle, EffectController controller)
+      : _angle = angle,
+        super(controller);
+
+  factory RotateEffect.to(double angle, EffectController controller) {
+    return _RotateToEffect(angle, controller);
+  }
+
+  /// The magnitude of the effect: how much the target should turn as the
+  /// progress goes from 0 to 1.
+  double _angle;
+
+  /// This variable keeps track of the effect's `progress` from the last time
+  /// that the `apply()` method was called.
+  double _lastProgress = 0;
+
+  @override
+  void apply(double progress) {
+    final dProgress = progress - _lastProgress;
```

Comment:
Will this way not create a larger rounding error until the effect is done? Instead of looking at how much the current progress should be converted to an angle?

Reply:
I consider it highly unlikely. A double has 16 digits of precision in its mantissa. For an average 1-second effect at 100fps, in the worst-case scenario we'd still retain 14 digits of precision. That's more than enough for a typical game.

Typically, the loss-of-precision occurs with floating points when adding and then subtracting quantities of significantly different magnitudes (such as `A + a - A`, where `A` is much larger than `a`), however this is not the scenario that we have here.

Reply:
Airtight, as long as we have tests for it, because I remember having problems with precision in the other effects system so that it was 1 pixel off when it ended.